src/api/java/com/minecolonies/api/colony/requestsystem/requestable/Weapon.java

```diff
@@ -0,0 +1,80 @@
+package com.minecolonies.api.colony.requestsystem.requestable;
+
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Class used to represent Weapons Inside the request system.
+ */
+public class Weapon
+{
+
+    @NotNull
+    private final WeaponType type;
+
+    @NotNull
+    private final Integer minLevel;
+
+    @NotNull
+    private final Integer maxLevel;
+
+    @NotNull
+    private final ItemStack result;
+
+    public Weapon(@NotNull final WeaponType type, @NotNull final Integer minLevel, @NotNull final Integer maxLevel)
+    {
+        this(type, minLevel, maxLevel, ItemStack.EMPTY);
+    }
+
+    public Weapon(@NotNull final WeaponType type, @NotNull final Integer minLevel, @NotNull final Integer maxLevel, @NotNull final ItemStack result)
+    {
+        this.type = type;
+        this.minLevel = minLevel;
+        this.maxLevel = maxLevel;
+        this.result = result;
+    }
+
+    /**
+     * The weapon type of the Weapon requested
+     *
+     * @return The weapon type of the weapon requested.
+     */
+    @NotNull
+    private WeaponType getType()
```

Comment:
Oeps IDEA autoconstructor seems to have made this private. This should be public.